pkg/cache/v3/simple.go

```diff
@@ -419,7 +432,13 @@ func (cache *snapshotCache) CreateDeltaWatch(request *DeltaRequest, state stream
 			cache.log.Errorf("failed to compute version for snapshot resources inline, waiting for next snapshot update")
 			delayedResponse = true
 		}
-		delayedResponse = respondDelta(request, value, state, snapshot, cache.log) == nil
+		response, err := respondDelta(context.Background(), request, value, state, snapshot, cache.log)
+		if err != nil {
+			cache.log.Errorf("failed to respond with delta response, waiting for next snapshot update: %s", err)
+			delayedResponse = true
```

Comment:
I think this statement is useless because we override delayedResponse on line 441 unconditionally.

Reply:
Oh I think you might be right. That's always going to execute regardless if we hit any of those errors